### ISSUE TITLE
SR012 when combining BIND with SERVICE

### ISSUE THREAD

**Issue by JervenBolleman** [openlink/virtuoso-opensource#244]
``` sparql
PREFIX up:<http://purl.uniprot.org/core/> 

PREFIX rdfs:<http://www.w3.org/2000/01/rdf-schema#> 
SELECT *
WHERE
{
    ?uniprot a up:Protein ;
             up:mnemonic ?protein_name ;
             rdfs:seeAlso ?db .
    ?updrugbank up:database <http://purl.uniprot.org/database/DrugBank> .
    BIND(iri(concat("http://bio2rdf.org/drugbank:DB00232", substr(str(?updrugbank), 34))) as ?bio2rdfdrugbank)
        SERVICE <http://drugbank.bio2rdf.org/sparql> {
      ?bio2rdfdrugbank <http://purl.org/dc/terms/title> ?title.
    }
}
```

SR012: Function aref needs a string or an array as argument 1, not an arg of type DB_NULL (204)

**Comment by HughWilliams** [comment]
Hi Jerven, I have been able to recreate against our LOD Cloud Server instance and have reported to development to look into:

http://lod.openlinksw.com/sparql?default-graph-uri=&qtxt=PREFIX+up%3A%3Chttp%3A%2F%2Fpurl.uniprot.org%2Fcore%2F%3E+%0D%0A%0D%0APREFIX+rdfs%3A%3Chttp%3A%2F%2Fwww.w3.org%2F2000%2F01%2Frdf-schema%23%3E+%0D%0ASELECT+*%0D%0AWHERE%0D%0A%7B%0D%0A++++%3Funiprot+a+up%3AProtein+%3B%0D%0A+++++++++++++up%3Amnemonic+%3Fprotein_name+%3B%0D%0A+++++++++++++rdfs%3AseeAlso+%3Fdb+.%0D%0A++++%3Fupdrugbank+up%3Adatabase+%3Chttp%3A%2F%2Fpurl.uniprot.org%2Fdatabase%2FDrugBank%3E+.%0D%0A++++BIND%28iri%28concat%28%22http%3A%2F%2Fbio2rdf.org%2Fdrugbank%3ADB00232%22%2C+substr%28str%28%3Fupdrugbank%29%2C+34%29%29%29+as+%3Fbio2rdfdrugbank%29%0D%0A++++++++SERVICE+%3Chttp%3A%2F%2Fdrugbank.bio2rdf.org%2Fsparql%3E+%7B%0D%0A++++++%3Fbio2rdfdrugbank+%3Chttp%3A%2F%2Fpurl.org%2Fdc%2Fterms%2Ftitle%3E+%3Ftitle.%0D%0A++++%7D%0D%0A%7D&format=text%2Fhtml&CXML_redir_for_subjs=121&CXML_redir_for_hrefs=&timeout=30000&debug=on

**Comment by JervenBolleman** [comment]
Still open in 7.2

**Comment by HughWilliams** [comment]
Have pinged development for an updated on this ...

**Comment by Alain-Gateau** [comment]
I have the exact same problem when running a fed query from our NeXtprot endpoint:

PREFIX :http://nextprot.org/rdf#
PREFIX annotation:http://nextprot.org/rdf/annotation/
PREFIX context:http://nextprot.org/rdf/context/
PREFIX cv:http://nextprot.org/rdf/terminology/
PREFIX db:http://nextprot.org/rdf/db/
PREFIX dc:http://purl.org/dc/elements/1.1/
PREFIX dcterms:http://purl.org/dc/terms/
PREFIX entry:http://nextprot.org/rdf/entry/
PREFIX evidence:http://nextprot.org/rdf/evidence/
PREFIX foaf:http://xmlns.com/foaf/0.1/
PREFIX gene:http://nextprot.org/rdf/gene/
PREFIX identifier:http://nextprot.org/rdf/identifier/
PREFIX isoform:http://nextprot.org/rdf/isoform/
PREFIX mo:http://purl.org/ontology/mo/
PREFIX ov:http://open.vocab.org/terms/
PREFIX owl:http://www.w3.org/2002/07/owl#
PREFIX publication:http://nextprot.org/rdf/publication/
PREFIX rdf:http://www.w3.org/1999/02/22-rdf-syntax-ns#
PREFIX rdfs:http://www.w3.org/2000/01/rdf-schema#
PREFIX sim:http://purl.org/ontology/similarity/
PREFIX source:http://nextprot.org/rdf/source/
PREFIX term:http://nextprot.org/rdf/terminology/
PREFIX xref:http://nextprot.org/rdf/xref/
PREFIX xsd:http://www.w3.org/2001/XMLSchema#
PREFIX up:http://purl.uniprot.org/core/
PREFIX taxon:http://purl.uniprot.org/taxonomy/
PREFIX rdf:http://www.w3.org/1999/02/22-rdf-syntax-ns#
PREFIX rdfs:http://www.w3.org/2000/01/rdf-schema#
PREFIX skos: http://www.w3.org/2004/02/skos/core#
​
select distinct ?entry ?ac ?mnemo where {
  ?entry :isoform / :binaryInteraction / :interactant ?interactant.
  ?entry skos:exactMatch ?skosuniprotid .
  ?interactant a :Xref. 
  ?interactant :accession ?ac.  
  #bind(str(substr(?ac,1,6)) as ?strac).
  #BIND (IRI(CONCAT("http://purl.uniprot.org/uniprot/",?strac)) as ?uniprotid)
   SERVICE http://beta.sparql.uniprot.org/sparql {
        ?skosuniprotid a up:Protein.
        ?skosuniprotid up:mnemonic ?mnemo .
     }
}
limit 3

If I uncomment only the first BIND then it results in a Error SR012, whether or not the bound
value is used in the service...
If I try to do the binding within the service instead it wont work either

**Comment by markwilkinson** [comment]
+1 - identical problem... still...

**Comment by HughWilliams** [comment]
@markwilkinson: Is this an issue for you also ? 
I have prompted development for the status of a fix for this issue as it was recreated locally ...

**Comment by markwilkinson** [comment]
Yes, it's a problem (new installation of Virtuoso).  The symptoms are 
~identical to those described above.  BIND(CONCAT(IRI)) can't be used to 
create an IRI for a federated query.

M

On 09/23/2015 11:19 AM, HughWilliams wrote:

> @markwilkinson https://github.com/markwilkinson: Is this an issue 
> for you also ?
> I have prompted development for the status of a fix for this issue as 
> it was recreated locally ...
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/openlink/virtuoso-opensource/issues/244#issuecomment-142539002.

## 

## 

Mark Wilkinson
Madrid, Spain
